federation/pkg/kubefed/init/init_test.go

```diff
@@ -77,6 +77,7 @@ func TestInitFederation(t *testing.T) {
 		image              string
 		etcdPVCapacity     string
 		expectedErr        string
+		dryRun             string
```

Comment:
`invalid-run` is a tricky thing to test as a whole. The right way to test it is to test each individual function in the code that gets called from `initFederation()` function. That was also one of the reasons why I asked you to push down `dryRun` to `waitForLoadBalancerAddress()` so that you can test that too.

If you actually decide going in that direction, it makes sense to make this a bool. But I will leave it to you to decide. I don't have a strong inclination towards one or the other.